subprojects/snapshots/src/integTest/groovy/org/gradle/internal/vfs/VirtualFileSystemRetentionIntegrationTest.groovy

```diff
@@ -174,6 +175,40 @@ class VirtualFileSystemRetentionIntegrationTest extends AbstractIntegrationSpec
         executedAndNotSkipped ":compileJava", ":classes", ":run"
     }
 
+    def "incubating message is shown for retention"() {
+        buildFile << """
+            apply plugin: "java"
+        """
+        def incubatingMessage = "Virtual File System Retention is an incubating feature"
+
+        when:
+        withRetention().run("assemble")
+        then:
+        outputContains(incubatingMessage)
+
+        when:
+        run("assemble")
+        then:
+        outputDoesNotContain(incubatingMessage)
+    }
+
+    def "incubating message is shown for partial invalidation"() {
```

Comment:
Should this be in a separate test class for partial invalidation instead?

Reply:
Is in a separate class now.